src/components/Admins/index.js

```diff
@@ -25,81 +32,97 @@ const Admins = () => {
   ];
 
   useEffect(() => {
-    async function fetchAdmins() {
-      try {
-        setLoading(true);
-        const response = await fetch(`${URL}`);
-        const { message, data, error } = await response.json();
-        if (!error) {
-          setBothDatas(data);
-        } else {
-          throw new Error(message);
-        }
-      } catch (error) {
-        console.log('error', error);
-      } finally {
-        setLoading(false);
-      }
+    if (!admins.length) {
+      dispatch(getAdmins());
+    }
+    if (error) {
+      openModal();
     }
-    fetchAdmins();
-  }, []);
+  }, [error]);
 
-  function handleDeleteAdmin(id) {
-    setDeleteId(id);
-    setIsOpen(true);
-  }
+  const handleSetAdmin = (id) => {
+    dispatch(setAdmin(id));
+    history.push('/admins/form');
+  };
 
-  function setBothDatas(data) {
-    setData(data);
-    setUntouchedData(data);
-  }
+  const buttonDelete = (id) => {
+    setAdminId(id);
+    openModal();
+  };
 
-  function search(value) {
+  const search = (value) => {
     setSearchQuery(value);
-    const result = untouchedData.filter((employee) => employee._id.includes(value));
-    setData(result);
-  }
+    setFilteredList(
+      admins.filter((item) => item.firstName.toLowerCase().includes(value.toLowerCase()))
+    );
+  };
 
-  async function deleteAdmin(id) {
-    const response = await fetch(`${URL}/${id}`, { method: 'DELETE' });
-    const responseJson = await response.json();
-    if (responseJson.error) {
-      alert('error');
-    } else {
-      setBothDatas(data.filter((item) => item._id !== id));
-      setIsOpen(false);
-    }
+  const openModal = () => {
+    setIsOpen(true);
+  };
+
+  const closeModal = () => {
+    setIsOpen(false);
+  };
+
+  if (loading) {
+    return (
+      <div className={styles.loading}>
+        <LoadingScreen />
+      </div>
+    );
   }
 
   return (
     <section className={styles.container}>
-      {loading ? (
+      <Modal
+        modalTitle={'Admins'}
+        isOpen={isOpen}
+        handleClose={() => {
+          openModal();
+        }}
```

Comment:
You could use here the reference of `closeModal` because the prop is called handleClose 😅 
```suggestion
        handleClose={closeModal}
```